arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceOptionsViewController.swift

```diff
@@ -0,0 +1,280 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+protocol ConfigureSubnetworkTraceOptionsViewControllerDelegate: AnyObject {
+    func optionsViewController(_ controller: ConfigureSubnetworkTraceOptionsViewController, didCreate: AGSUtilityTraceConditionalExpression)
+}
+
+class ConfigureSubnetworkTraceOptionsViewController: UITableViewController {
+    // MARK: Storyboard views
+    
+    /// The cell for attribute options.
+    @IBOutlet var attributesCell: UITableViewCell!
+    /// The cell for comparison operator options.
+    @IBOutlet var comparisonCell: UITableViewCell!
+    /// The cell for value to compare with.
+    @IBOutlet var valueCell: UITableViewCell!
+    /// A button to add the conditional expression to the trace configuration.
+    @IBOutlet var doneBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties
+    
+    /// A delegate to notify other view controllers.
+    weak var delegate: ConfigureSubnetworkTraceOptionsViewControllerDelegate?
+    
+    /// An array of possible network attributes.
+    var possibleAttributes = [AGSUtilityNetworkAttribute]()
+    
+    /// The attribute selected by the user.
+    var selectedAttribute: AGSUtilityNetworkAttribute? {
+        didSet {
+            // Set the selected attribute name.
+            attributesCell.detailTextLabel?.text = selectedAttribute?.name
+            // Reset the selected value.
+            selectedValue = nil
+            valueCell.detailTextLabel?.text = nil
+            updateCellStates()
+        }
+    }
+    /// The comparison selected by the user.
+    var selectedComparison: AGSUtilityAttributeComparisonOperator? {
+        didSet {
+            if let selectedComparisonString = selectedComparison?.title {
+                comparisonCell.detailTextLabel?.text = selectedComparisonString
+            } else {
+                comparisonCell.detailTextLabel?.text = nil
+            }
+            doneBarButtonItem.isEnabled = selectedComparison != nil && selectedValue != nil
+        }
+    }
+    /// The value selected by the user.
+    var selectedValue: Any? {
```

Comment:
Should the type of this be `AGSCodedValue`?

Reply:
No, `selectedValue` can also be a number since the user is able to input a custom value depending on which attribute is chosen.